Tests/L0/Gradle/_suite.ts

```diff
@@ -1265,17 +1250,74 @@ describe('gradle Suite', function () {
                 assert(tr.ran(gradleWrapper + ' build -I /Gradle/CodeAnalysis/pmd.gradle'), 'Ran Gradle with PMD');
                 assert(tr.stdout.indexOf('task.addattachment type=Distributedtask.Core.Summary;name=Code Analysis Report') > -1,
                     'should have uploaded a Code Analysis Report build summary');
-                assert(tr.stdout.indexOf('artifact.upload artifactname=Code Analysis Results;') > -1,
-                    'should have uploaded code analysis build artifacts');
+                assert(tr.stdout.indexOf('artifact.upload artifactname=Code Analysis Results;') < 0,
+                    'should not have uploaded code analysis build artifacts');
 
                 assertCodeAnalysisBuildSummaryDoesNotContain(testStgDir, 'Checkstyle found no violations.');
                 assertCodeAnalysisBuildSummaryContains(testStgDir, 'PMD found no violations.');
 
-                var codeAnalysisStgDir: string = path.join(testStgDir, '.codeAnalysis', 'CA');
+                // There were no files to be uploaded - the CA folder should not exist
+                var codeAnalysisStgDir: string = path.join(testStgDir, '.codeAnalysis');
+                assertFileDoesNotExistInDir(codeAnalysisStgDir, 'CA');
 
-                // Test files were copied for module "root", build 14
-                assertFileDoesNotExistInDir(codeAnalysisStgDir, '/root/14_main_Checkstyle.xml');
-                assertFileDoesNotExistInDir(codeAnalysisStgDir, '/root/14_main_PMD.xml');
+                done();
+            })
+            .fail((err) => {
+                console.log(tr.stdout);
+                console.log(tr.stderr);
+                console.log(err);
+                done(err);
+            });
+
+        // Clean up
+        cleanTempDirsForCodeAnalysisTests();
+    });
+
+    it('Gradle with code analysis - Does not upload artifacts if code analysis reports were empty', function (done) {
```

Comment:
https://github.com/Microsoft/vsts-tasks/pull/2746/files/6b6f629a680cfd457431d0eece5b29a6e326e68f#diff-260cb51b9bd3175ad8b53a45ae5c59caR1222 

The above test asserts that when only one tool is enabled and three empty reports are found, that an artifact is uploaded and the build summary contains only the line for the tool which is enabled.